src/Microsoft.Identity.Client/ITokenCache.cs

```diff
@@ -53,6 +54,24 @@ public interface ITokenCache
         /// <param name="beforeWrite">Delegate set in order to prepare the cache serialization</param>
         void SetBeforeWrite(TokenCacheCallback beforeWrite);
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="beforeAccess"></param>
+        void SetAsyncBeforeAccess(Func<TokenCacheNotificationArgs, Task> beforeAccess);
```

Comment:
>Async [](start = 16, length = 5)

Suggestion Async to be appended instead? (all functions)

Reply:
I didn't want these to end in Async because that implies the method itself is async, which it isn't.  The pattern for async methods is to suffix them with Async.  So this is setting the "async before access callback" which is why I named it that way.

Reply:
is there an official calling convention for these things? At this point in the day I feel it makes more sense to flip it, e.g. when looking for a mthod doing x then I can easily see xAsync as well. Understand your argument as well. discoverability get's higher though imo if flipping things around.

---
In reply to: [281857006](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1088#discussion_r281857006) [](ancestors = 281857006)

Reply:
No, there isn't.  I'm not married to the names as they are, and you make a good point about discoverability.  But it is a common convention that methods ending in Async return a task.  Perhaps SetBeforeAccessWithAsync() (note the word With)?